doc/parallel.rst

```diff
@@ -128,6 +128,35 @@ parallelism automatically tries to maintain and reuse a pool of workers
 by it-self even for calls without the context manager.
 
 
+Avoiding over-subscription of CPU ressources
+============================================
+
+The computation parallelism relies on the usage of multiple CPUs to perform the
+operation simultaneously. When using more processes than the number of CPU on
+a machine, the performance of each process is degraded as there is less
+computational power available for each process. Moreover, when many processes
+are running, the time taken by the OS scheduler to switch between them can
+further hinder the performance of the computation. It is generally better to
+avoid using more ``Process/Thread`` than the number of CPU on a machine.
```

Comment:
avoid using significantly more processes or threads than the number of CPUs on a machine.